src/dhwcalc.cpp

```diff
@@ -3515,57 +3518,49 @@ RC HPWHLINK::hw_DoHour(		// hourly HPWH calcs
 // Does HPWH setup etc that need not be done subhourly
 // returns RCOK iff success
 {
-	RC rc = RCOK;
-
 	if (Top.tp_isBegMainSim)
 		hw_balErrCount = 0;
 
 	// setpoint temp: ws_tUse has hourly variability
 	//   some HPWHs (e.g. SANCO2) have fixed setpoints, don't attempt
-	if (!hw_pHPWH->isSetpointFixed())
-	{	double tSetpointMax;
-		std::string whyNot;		// HPWH explanatory text, ignored
-		bool bSPP = hw_pHPWH->isNewSetpointPossible(tSetpoint, tSetpointMax, whyNot, HPWH::UNITS_F);
-		// silently limit to max acceptable
-		//   if HPWH has resistance, max = 212
-		float tSetpointX = bSPP ? tSetpoint : tSetpointMax;
-		hw_pHPWH->setSetpoint(tSetpointX, HPWH::UNITS_F);
-	}
-
-	// retrieve resulting setpoint after HPWH restrictions
-	tSetpoint = hw_pHPWH->getSetpoint(HPWH::UNITS_F);
-	if (hw_tHWOut == 0.f)
-		hw_tHWOut = tSetpoint;		// initial guess for HW output temp
-									//   updated every substep with nz draw
-
-	// tank temp initialization
-	if (!hw_tankTempSet)
-	{	// initialize tank temp on 1st call
-		//   must be done after setting HPWH setpoint (=ws_tSetpoint)
-		//   (ws_tSetpoint may be expression)
-		if (tankTInit != nullptr)
-		{
-			std::vector<double> vTankTInit;
-			vTankTInit.assign(tankTInit, tankTInit + 12);
-			if (hw_pHPWH->setTankLayerTemperatures(vTankTInit, HPWH::UNITS_F))
-				rc |= RCBAD;
-		}
-		else
-		{
-			if (hw_pHPWH->resetTankToSetpoint())
-				rc |= RCBAD;
-		}
-		hw_tankTempSet = true;
-	}
+    if (!hw_pHPWH->isSetpointFixed()) {
+        double tSetpointMax;
+        std::string whyNot; // HPWH explanatory text, ignored
+        bool bSPP = hw_pHPWH->isNewSetpointPossible(tSetpoint, tSetpointMax,
+                                                    whyNot, HPWH::UNITS_F);
+        // silently limit to max acceptable
+        //   if HPWH has resistance, max = 212
+        float tSetpointX = bSPP ? tSetpoint : tSetpointMax;
+        hw_pHPWH->setSetpoint(tSetpointX, HPWH::UNITS_F);
+    }
+
+    // retrieve resulting setpoint after HPWH restrictions
+    tSetpoint = hw_pHPWH->getSetpoint(HPWH::UNITS_F);
+    if (hw_tHWOut == 0.f)
+        hw_tHWOut = tSetpoint; // initial guess for HW output temp
+                                //   updated every substep with nz draw
+
+    // tank temp initialization
+    if (!hw_tankTempSet) { // initialize tank temp on 1st call
```

Comment:
Accidental formatting of comment? Should be on next line aligned with other comments.